syntax = "proto3";

service EmbeddingService {
    rpc EmbedImage (ImageEmbeddingRequest) returns (Embedding) {}
    rpc EmbedText (TextEmbeddingRequest) returns (Embedding) {}
}

message ImageEmbeddingRequest {
    bytes image_bytes = 1;
}



message TextEmbeddingRequest {
    string text = 1;
}

message Embedding {
    repeated float values = 1;
    int32 dim = 2;
}